File: 3300-minimum-element-after-replacement-with-digit-sum/3300-minimum-element-after-replacement-with-digit-sum.java
class Solution {
    public int minElement(int[] nums) {
        int min = Integer.MAX_VALUE;
        for(int num:nums){
            int sum = 0;
            while(num>0){
                int rem = num%10;
                sum += rem;
                num /= 10;
            }
            min = Math.min(min,sum);
        }
        return min;
        
    }
}

File: 3300-minimum-element-after-replacement-with-digit-sum/3300-minimum-element-after-replacement-with-digit-sum.py
class Solution:
    def minElement(self, nums: List[int]) -> int:
        sum_of_nums = []
        for val in nums:
            sum=0
            while val>0:
                rem = val%10
                sum+=rem
                val //= 10
            sum_of_nums.append(sum)
        return min(sum_of_nums)

        